provider/oracle/instance.go

```diff
@@ -84,7 +83,11 @@ func newInstance(params response.Instance, env *OracleEnviron) (*oracleInstance,
 // Id is defined on the instance.Instance interface.
 func (o *oracleInstance) Id() instance.Id {
 	if o.machine.Name != "" {
-		return instance.Id(o.machine.Name)
+		name := strings.Split(o.machine.Name, "/")
```

Comment:
can you please add a comment here describing the structure of the name, which justifies the code below? e.g. something like

> Machine names have the format "/Compute-identity_domain/user/name_you_specify/id", where id is a unique ID generated by the cloud service.